openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -0,0 +1,167 @@
+            'pre_requisite_courses'
+        ]
+        for attribute_name in fields_to_test:
+            course_value = getattr(course, attribute_name)
+            cache_miss_value = getattr(course_overview_cache_miss, attribute_name)
+            cache_hit_value = getattr(course_overview_cache_hit, attribute_name)
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Test if return values all methods are equal between the three objects
+        methods_to_test = [
+            'clean_id',
+            'has_ended',
+            'has_started',
+            'start_datetime_text',
+            'end_datetime_text',
+            'may_certify',
+        ]
+        for method_name in methods_to_test:
+            course_value = getattr(course, method_name)()
+            cache_miss_value = getattr(course_overview_cache_miss, method_name)()
+            cache_hit_value = getattr(course_overview_cache_hit, method_name)()
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Other values to test
+        others_to_test = [(
+            course_image_url(course),
+            course_overview_cache_miss.course_image_url,
+            course_overview_cache_hit.course_image_url
+        ), (
+            get_active_web_certificate(course) is not None,
+            course_overview_cache_miss.has_active_web_certificates,
+            course_overview_cache_hit.has_active_web_certificates
+
+        )]
+        for (course_value, cache_miss_value, cache_hit_value) in others_to_test:
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+    @ddt.data(*itertools.product(
+        [
+            {
+                "static_asset_path": "/my/cool/path",
+                "display_name": "Test Course",
+                "start": YESTERDAY,
+                "end": TOMORROW,
+                "pre_requisite_courses": ['course-v1://edX+test1+run1', 'course-v1://edX+test2+run1']
+            },
+            {}
+        ],
+        [ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split]
+    ))
+    @ddt.unpack
+    def test_course_overview_behavior(self, course_kwargs, modulestore_type):
+        """Tests if CourseOverviews and CourseDescriptors behave the same
+        by comparing pairs of them given a variety of scenarios.
+
+        Args:
+            course_kwargs (dict): kwargs to be passed to course constructor
+            modulestore_type (ModuleStoreEnum.Type)
+            is_user_enrolled (bool)
+        """
+
+        course = CourseFactory.create(
+            course="test_course",
+            org="edX",
+            default_store=modulestore_type,
+            **course_kwargs
+        )
+        self.check_course_overview_against_course(course)
+
+    def test_course_overview_cache_invalidation(self):
```

Comment:
Please add calls to `check_mongo_calls` to validate the performance improvement via caching.